packages/e2e-test-utils/src/wait-for-animation.js

```diff
@@ -0,0 +1,10 @@
+/**
```

Comment:
Understanding the desire that this is introduced to help unblock the pull request, I hesitate on whether this should be considered an "unstable" API or not. I think as proposed it's at least doing well enough to not assume that it operates on some timer, since I think an implementation leveraging [`transitionend`](https://developer.mozilla.org/en-US/docs/Web/Events/transitionend) is probably a worthwhile exploration for further improvement. But the bigger issue to me is forcing it to be a conscious part of someone's mind in authoring tests, and whether that's strictly necessary. It seems far too prone to being forgotten and causing unexpected test failures, though if they're obvious enough and consistent failures, perhaps they're not truly "unexpected" and it's okay.

In short, I wonder if it would be a good idea to mark this as unstable, or perhaps not if we think that even if it be part of a public API (and usable directly) that we could find some other way to "automate" this on behalf of a test author in the future.

Reply:
Other possibilities to avoid it being a conscious part of the developer test authoring workflow is to ensure that most of the interactions which involve an animation are sufficiently abstracted. So rather than a test including:

```js
await page.click( 'button[aria-label="More options"]' );
await waitForAnimation();
```

We'd see something like `await openMoreOptions();` which handles the animation implementation detail internally.

Reply:
I do agree we should "automate" this on behalf of the test author behind utilities like:

 - insert block
 - click menu item
 - switch to editor mode
...

at that point we could hide it entirely from the public API

A lot of tests need to be updated though, and I don't think that's the PR for it.

Reply:
@aduth Looking at the tests, I don't think we can entirely abstract all the interactions without creating very specific functions. One example that comes to mind, is do we want to add an  `openFontSizePicker` utility where it's very specific to a given a block in a given state?

Reply:
> A lot of tests need to be updated though, and I don't think that's the PR for it.

As long as there's some reasonably clear path forward (and with at least a nod to the fact we're adding technical debt) I'm fine to not let it block the work here. It would be good to have an issue describing the future planned work.

Reply:
I honestly don't have a clear path forward on my mind. I'd love thoughts. 

Abstractions over the generic use-cases (block menus, global menus, inserter) are a good solution but what about the very specific use-cases?

Reply:
> One example that comes to mind, is do we want to add an `openFontSizePicker` utility where it's very specific to a given a block in a given state?

No, probably not. But then it resurfaces the problem of imposing the need to be aware of a delay to the test author. My original intent with raising the point was in trying to see through the implications, and if not to solve it in place, have some idea for a future goal. To me, that future goal ideally wouldn't involve `waitForAnimation` existing, hence why I'd considered whether to mark it as unstable.

The alternatives I'd have in mind which bypass the need for the function are rough, and in many ways similarly flawed / technically challenging:

- Not have the transitions exist for the tests. It's still an issue that these are going to add measurable delays to the runtime of every build.
- In some way, every `transitionstart` which ever occurs on a page is monitored such that when a test includes an interaction (e.g. click) which either triggers / follows (unsure which yet) a transition, the following action would be delayed automatically.

I'm also open to saying that accepting that this needs to be consciously considered as part of writing an end-to-end tests. I'm sure there's an argument that these "steps" follow the user's actions, which includes that when a user clicks a button, they may have to wait for the animation.

tl;dr: I have concerns, and I wanted them to be noted, but I also think this is good enough in its current state to be able to move forward on.

Reply:
Thanks for the thoughts, I'm personally not concerned about stable/unstable API since it's just a test utility that said, I'll follow up with an issue with these great thoughts.

I just had an idea probably similar to your first point here: maybe in the tests we load a specific stylesheet forcing all animations to be very quick (unnoticeable by tests) like 1ms. 

 - It ensures we're testing the existence of the animation (like prod)
 - It ensures the tests are stable.

Reply:
> I'm personally not concerned about stable/unstable API since it's just a test utility

Speaking of breaking changes which would require a major version bump to `@wordpress/e2e-test-utils`: We should include a note (enhancement, minor bump) for the addition of `waitForAnimation` and ideally document it somewhere if we're doing that for these utilities (I'd suspect not).

Reply:
Actually that package hasn't been released yet. It feels weird to add just `waitForAnimation` to the changelog where it only includes a single entry `Initial release`.

Good call though, I added the `Animate` component to the components changelog.

Reply:
> It feels weird to add just `waitForAnimation` to the changelog where it only includes a single entry `Initial release`.

Oh, I didn't realize it hadn't been released yet. It's fine to just leave it then as "Initial release".